TodoList/Src/TodoList.Contracts/DI/DependencyResolutionFailedException.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace TodoList.Contracts.DI
+{
+    public class DependencyResolutionFailedException : Exception
+    {
+        public DependencyResolutionFailedException()
+        {
+        }
+
+        public DependencyResolutionFailedException(string message) : base(message)
```

Comment:
Why I need to remove the public constructor which are standard in Exception (MyException inherit from Exception). I know that I don't need them now and probably anytime in this project, but isn't that something as common public API which you expected from Exception?

I also found this [StackOverflow answer](https://stackoverflow.com/a/931535/10290539).

Reply:
Well, you answered yourself: It is (and won't be) use - [YAGNI](https://martinfowler.com/bliki/Yagni.html) helps a great deal in simplifying codebase and helping (other) developers to use it as intended and designed.

> …but isn't that something as common public API which you expected from Exception?

No😃

I will also quote from the `SO` you sent 😺:
> This is a warning, not a requirement.

I agree and I believe it to be well-placed notice - it is a really nice application of the principle of least surprise, however, you need to ask "Who are your 'users'?" and "What is their relation to the system that the exception is offered by?"

> If you have a good reason to ignore the guideline, do so.

You have a perfect reason: This exception without a message and inner exception makes no sense. 

### In more detail:
Throwing the exception without inner one makes it meaningless as you have no clue what really happened, what type was not resolved (you might need an information on its full name, assembly it is loaded from, ... No string value will be complete even if it was in the message). It was your intention when you created the class to wrap the Unity's exception after all, right? 

Throwing the exception without message does not bring any value on the state of the system causing it or original author's intention - I believe it to be a code smell when a developer is not able to express the cause of him breaking the natural flow of their application.
You can use a message-less exception when prototyping something quick, but not something you and many different developers will attend to over year(s) - there is a nice [quote regards the general design of your code](https://stackoverflow.com/a/878436/1138663).
Basically, you want to provide them (and you) with most information in code that they can use (they won't be able to obtain it otherwise) when it is necessary to troubleshoot the code - assert and exception messages are one of a few tools you can use to do just that.

Reply:
Thank you for nice explanation. :simple_smile:

Reply:
3085dc2